### PR TITLE
adaptation join spec

### PR DESCRIPTION
**Pull request by elsdvlee**
## What's changed

Adapted the specification of joins

## Change checklist
- [x] updated ontology, where necessary
- [x] updated shapes, where necessary
- [x] added or updated test cases, where necessary
- [x] any TODOs have been turned into trackable issues and referenced where necessary

### CODE REVIEW

File: spec/section/joins.md [hunk 1]
```diff
@@ -1,116 +1,38 @@
...
-- at least one join condition property (`rml:joinCondition`), whose value is a [=join condition=] that describe which values are compared to join the two logical views.
-- at least one field property (`rml:field`), whose value is a [=field=] (`rml:Field`) . This field MAY only contain references to fields that exists in the parent logical view. 
+- exactly one parent logical view property (`rml:parentLogicalView`), whose value is a [=logical view=] (`rml:LogicalView`) that supplies the additional fields, fulfills the role of the <!-- TODO reference to core parent logical source when available-->[parent logical source]() in the <a data-cite="RML-Core#dfn-join-condition">join condition(s)</a> of the [=logical view join=], and is referred to as <dfn>parent logical view</dfn>.
+- at least one join condition property (`rml:joinCondition`), whose value is a <a data-cite="RML-Core#dfn-join-condition">join condition</a>.
+- at least one field property (`rml:field`), whose value is a [=field=] (`rml:Field`). This field MAY only contain references to fields that exists in the parent logical view.
```
Comment (pmaria) [inline spec/section/joins.md:8]:
```suggestion
- at least one field property (`rml:field`), whose value is a [=field=] (`rml:Field`). This field SHOULD only contain references to fields that exists in the parent logical view. 
```

File: spec/section/joins.md [hunk 2]
```diff
@@ -1,116 +1,38 @@
 ## Logical view joins {#viewjoins}
 
-A <dfn>logical view join</dfn> (`rml:LogicalViewJoin`) is an operation that extends the logical iteration of one logical view (the child logical view) with fields from another logical view (the parent logical view).
+A <dfn>logical view join</dfn> (`rml:LogicalViewJoin`) is an operation that extends the logical iteration of one logical view (the [=child logical view=]) with fields derived from  another logical view (the [=parent logical view=]).
```
Comment (pmaria) [inline spec/section/joins.md:3]:
```suggestion
A <dfn>logical view join</dfn> (`rml:LogicalViewJoin`) is an operation that extends the logical iteration of one logical view (the [=child logical view=]) with fields derived from another logical view (the [=parent logical view=]).
```

File: spec/section/joins.md [hunk 3]
```diff
@@ -1,116 +1,38 @@
...
-Els: or can we just refer to rml core and not specify any definitions here?
-</aside>
-<aside class="issue">
-Els: can we also optionally declare a join function here, to allow not only equijoins (default) but also other joins
-</aside>
-
 ### Join types {#dfn-join-type}
 
-A [=logical view=] (`rml:LogicalView`) MAY have one or more with join properties, specifying the join type, i.e. a [=left join=] and a [=inner join=].
+A [=logical view=] (`rml:LogicalView`) MAY have one or more join properties, whose value is a [=logical view join=].
```
Comment (pmaria) [inline spec/section/joins.md:18]:
```suggestion
A [=logical view=] (`rml:LogicalView`) MUST have zero or more join properties, whose value is a [=logical view join=]. 
```

File: spec/section/joins.md [hunk 4]
```diff
@@ -1,116 +1,38 @@
...
-If no match is found, the added field in that extended logical iteration contains a null value.
+The join property specifies the join type of the [=logical view join=], i.e. a [=left join=] and an [=inner join=]. 
 
-A <dfn>inner join</dfn> (`rml:innerJoin`) is the equivalent of an inner join in SQL.
-The logical iterations from the child logical view are extended with values from the parent logical view when a match is found that meets the join conditions.
-When more than one logical iteration in the parent logical view matches with a logical iteration in the child logical view, each match leads to an additional extended logical iteration.
-If no match is found for a logical iteration, the logical iteration is removed from the child logical view.
+A <dfn>left join</dfn> (`rml:leftJoin`) is the equivalent of a left (outer) join in SQL, where the [=child logical view=] is the left part of the join, and the [=parent logical view=] is the right part of the join. If any of the <a data-cite="RML-Core#dfn-join-condition">join condition</a> evaluates `false`, the fields from the [=logical view join=] in the extended logical iteration contain a null value.
```
Comment (pmaria) [inline spec/section/joins.md:24]:
```suggestion
A <dfn>left join</dfn> (`rml:leftJoin`) is the equivalent of a left (outer) join in SQL, where the [=child logical view=] is the left part of the join, and the [=parent logical view=] is the right part of the join. If any of the <a data-cite="RML-Core#dfn-join-condition">join conditions</a> evaluates to `false`, the fields from the [=logical view join=] in the extended logical iteration contain a null value.
```

File: spec/section/joins.md [hunk 5]
```diff
@@ -1,116 +1,38 @@
...
 
-<aside class="issue">
-Pano: If there are more than one join, what is the order of execution?
-</aside>
+An <dfn>inner join</dfn> (`rml:innerJoin`) is the equivalent of an inner join in SQL. If any of the <a data-cite="RML-Core#dfn-join-condition">join condition</a> evaluates `false`, the logical iteration is removed from the [=child logical view=].
```
Comment (pmaria) [inline spec/section/joins.md:26]:
```suggestion
An <dfn>inner join</dfn> (`rml:innerJoin`) is the equivalent of an inner join in SQL. If any of the <a data-cite="RML-Core#dfn-join-condition">join conditions</a> evaluates to `false`, the logical iteration is removed from the [=child logical view=].
```

File: spec/section/joins.md (hunk 3 above)
Comment (pmaria) [inline spec/section/joins.md:18]: This is also already stated in section 4. Logical Views. Ideally, we should only state it once, either here or in section 4.
If we need to reference "join properties" here, we should probably define the term.

**Review by pmaria — CHANGES_REQUESTED** [review]
Looking good, some minor comments.

File: spec/section/joins.md (hunk 3 above)
Comment (elsdvlee) [reply spec/section/joins.md]: See my new commit: https://github.com/kg-construct/rml-lv/pull/36/commits/63046b29f38144835c7e549f290f9cb8df3c3747.

File: spec/section/joins.md [hunk 6]
```diff
@@ -1,128 +1,43 @@
 ## Logical view joins {#viewjoins}
 
-A <dfn>logical view join</dfn> (`rml:LogicalViewJoin`) is an operation that extends the logical iteration of one logical view (the child logical view) with fields from another logical view (the parent logical view).
+A <dfn>logical view join</dfn> (`rml:LogicalViewJoin`) is an operation that extends the logical iteration of one logical view (the [=child logical view=]) with fields derived from another logical view (the [=parent logical view=]).
```
Comment (pmaria) [inline spec/section/joins.md:3]:
```suggestion
A <dfn>logical view join</dfn> (`rml:LogicalViewJoin`) is an operation that extends the logical iteration of one logical view (the [=child logical view=]) with fields derived from another logical view (the [=parent logical view=]).
```

File: spec/section/joins.md [hunk 7]
```diff
@@ -1,128 +1,43 @@
...
-Els: or can we just refer to rml core and not specify any definitions here?
-</aside>
-<aside class="issue">
-Els: can we also optionally declare a join function here, to allow not only equijoins (default) but also other joins
-</aside>
-
 ### Join types {#dfn-join-type}
 
-A [=logical view=] (`rml:LogicalView`) MAY have one or more with join properties, specifying the join type, i.e. a [=left join=] and a [=inner join=].
+The <dfn>join property</dfn> specifies the join type of the [=logical view join=], i.e. a [=left join=] and an [=inner join=].
```
Comment (pmaria) [inline spec/section/joins.md:20]:
```suggestion
The <dfn>join property</dfn> specifies the join type of the [=logical view join=], i.e. a [=left join=] or an [=inner join=].
```

**Review by pmaria — APPROVED** [review]
2 minor comments

**Pull request merged**